pkg/wharfapi/client.go

```diff
@@ -26,3 +32,71 @@ type Client struct {
 // Deprecated: This type has been renamed to Client and may be removed in a
 // future release.
 type WharfClient Client
+
+func (c Client) Get(path string, q url.Values) ([]byte, error) {
+	req, err := c.NewRequest(http.MethodGet, path, q, nil)
+	if err != nil {
+		return nil, err
+	}
+	return doRequest(req)
+}
+
+func (c Client) GetUnmarshal(path string, q url.Values, response interface{}) error {
+	bytes, err := c.Get(path, q)
+	if err != nil {
+		return err
+	}
+	return json.Unmarshal(bytes, response)
+}
+
+func (c Client) Post(path string, q url.Values, body []byte) ([]byte, error) {
+	req, err := c.NewRequest(http.MethodPost, path, q, body)
+	if err != nil {
+		return nil, err
+	}
+	return doRequest(req)
+}
+
+func (c Client) PostJSON(path string, q url.Values, body interface{}) ([]byte, error) {
+	bodyBytes, err := json.Marshal(body)
+	if err != nil {
+		return nil, err
+	}
+	return c.Post(path, q, bodyBytes)
+}
+
+func (c Client) PostJSONUnmarshal(path string, q url.Values, body interface{}, response interface{}) error {
+	responseBytes, err := c.PostJSON(path, q, body)
+	if err != nil {
+		return err
+	}
+	return json.Unmarshal(responseBytes, response)
+}
+
+func (c Client) Put(path string, q url.Values, body []byte) ([]byte, error) {
+	req, err := c.NewRequest(http.MethodPut, path, q, body)
+	if err != nil {
+		return nil, err
+	}
+	return doRequest(req)
+}
+
+func (c Client) PutJSON(path string, q url.Values, body interface{}) ([]byte, error) {
+	bodyBytes, err := json.Marshal(body)
+	if err != nil {
+		return nil, err
+	}
+	return c.Put(path, q, bodyBytes)
+}
+
+func (c Client) PutJSONUnmarshal(path string, q url.Values, body interface{}, response interface{}) error {
+	responseBytes, err := c.PutJSON(path, q, body)
+	if err != nil {
+		return err
+	}
+	return json.Unmarshal(responseBytes, response)
+}
+
+func (c Client) NewRequest(method, path string, q url.Values, body []byte) (*http.Request, error) {
```

Comment:
Actually don't see the need for any of these Get,Post,etc methods inside `client.go` to be exported